R/utils.R

```diff
@@ -167,10 +189,14 @@ setMethod("variantFilter",
 }
 
 
-.meanImputeFn <- function(geno, freq) {
+.meanImputeFn <- function(geno, freq, geno.coding="additive") {
     miss.idx <- which(is.na(geno))
     miss.var.idx <- ceiling(miss.idx/nrow(geno))
-    geno[miss.idx] <- 2*freq[miss.var.idx]
+    if (geno.coding == "additive") {
+        geno[miss.idx] <- 2*freq[miss.var.idx]
+    } else {
+        geno[miss.idx] <- freq[miss.var.idx]
```

Comment:
@smgogarten I actually don't think this is correct. I think we want to impute with `freq[[out.col]]/n.obs` -- i.e. the frequency of the "1" value. Unfortunately, that seems like it might mean greater modification to this function.

Reply:
Can we just get that frequency from `geno` directly as `freq <- colMeans(geno, na.rm=TRUE)`? We don't do that in the additive case because 1) we've just calculated frequency already and 2) sex chromosomes need special treatment, but I don't think either of those apply here.

Reply:
Yeah. That should be fine - just compute the `freq` vector in the else statement right here. 

Do sex chr need any special consideration for recessive or dominant coding? Females should be fine. For males, a carrier should definitely be a "1" for dominant (I think they would be); a carrier should probably(?) be a "1" for recessive, which they might not be? (I'm actually not sure).

Reply:
Males are generally coded as 0/2, in which case a carrier would be "1" for either dominant or recessive. We do have the option for a user to specify `male.diploid=FALSE`, in which case they are expected to be coded 0/1. So I think we need another `if` statement under recessive, where if `male.diploid=TRUE` then X chr variants for males get set to 1 if `geno == 1`.

Reply:
My thinking was the same as Adrienne's. For a recessive trait, carriers are not affected (or are less affected) because they still have a copy of the non-effect allele being expressed. For males on the X chromosome, if they have the effect allele they will exhibit the trait. (Hemophilia for example, which is caused by a recessive X chromosome mutation.) The additive/recessive/dominant coding reflects the biology of the trait, which the `male.diploid` flag only reflects the conventions used in the input VCF file.

Reply:
Works for me. We should just make sure the documentation is clear.

Reply:
There's even a word for it! Males are hemizygous on the X chromosome.